apps/src/calc/calc.js

```diff
@@ -199,26 +201,31 @@ Calc.init = function (config) {
     }
   };
 
-  var codeWorkspace = (
-    <CodeWorkspace
-      localeDirection={studioApp.localeDirection()}
-      editCode={!!level.editCode}
-      readonlyWorkspace={!!config.readonlyWorkspace}
-    />
-  );
+  // Push initial level properties into the Redux store
+  studioApp.reduxStore.dispatch(setInitialLevelProps({
+    localeDirection: studioApp.localeDirection(),
```

Comment:
I agree that `level` is misleading. Really this is just a collection of properties that we initialize, and don't expect to ever change. It seems reasonable to me to have them all be part of the same concept, and have that concept be something other than level. Maybe `getState().initialProps.foo` or `configConstants` or ... ?